topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/NamedThreadFactory.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.aries.rsa.topologymanager;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class NamedThreadFactory implements ThreadFactory {
+    /**
+     * Counter used when constructing unique Thread names
+     */
+    private final AtomicInteger count = new AtomicInteger(0);
+    /**
+     * Pattern used when constructing Thread names. This pattern must include the sequence
+     * %d which will be used to replace with a Thread count.
+     */
+    private final String namePattern;
+
+    /**
+     * @param namePattern Name pattern to use when constructing Threads
+     */
+    public NamedThreadFactory(String namePattern) {
+        if (namePattern == null || !namePattern.contains("%d")) {
```

Comment:
Should it also be checked that `%d` occurs only once?

Reply:
Could be, though it would only prevent a 'raw' %d being printed. My main concern was that the absence of any %d would result in a MissingFormatArgumentException.
If there is a preference to detect an excess of %d's I would be happy to update the PR,

Reply:
Why do we check this? We only call NamedThreadFactory with a constant string. So we already know at build time if the string is ok.

Reply:
I think this was a 'just in case' matter, when these were to be used elsewhere.
@cschneider do you prefer that both null and %d be removed?

Reply:
With the change to use a class to construct the NamedThreadFactory, resoles the various concerns raised here. Resolving this conversation.